qcodes/dataset/data_set.py

```diff
@@ -1254,6 +1254,14 @@ def _load_to_xarray_dataset(self, data: ParameterData) -> "xr.Dataset":
                 paramspec_dict = self.paramspecs[str(dim)]._to_dict()
                 xrdataset.coords[str(dim)].attrs.update(paramspec_dict.items())
 
+        self._add_metadata_to_xarray(xrdataset)
```

Comment:
but this way wouldn't we have duplicated snapshot in xrdataset and all it's xrdatarrays ?